components/Agents/Agents.tsx

```diff
@@ -18,73 +92,88 @@ const Agents = () => {
-          />
-          <FunnelCard
-            funnelName="instagram"
-            icon="white-instagram"
-            onClick={() => handleClick("instagram")}
-          />
-        </div>
+        ))}
+        {extraTags.length > 0 && !showAllTags && (
+          <button
+            type="button"
+            onClick={() => setShowAllTags(true)}
+            className="ml-2 text-xs text-gray-500 hover:underline bg-transparent border-none shadow-none p-0 flex items-center"
+            style={{ background: 'none', border: 'none', fontWeight: 400 }}
+          >
+            More
+            <svg width="12" height="12" viewBox="0 0 20 20" fill="none" className="ml-1">
+              <title>Show more tags</title>
+              <path d="M6 8l4 4 4-4" stroke="currentColor" strokeWidth="2" strokeLinecap="round" strokeLinejoin="round"/>
+            </svg>
+          </button>
+        )}
+        {showAllTags && (
+          <button
+            type="button"
+            onClick={() => setShowAllTags(false)}
+            className="ml-2 text-xs text-gray-500 hover:underline bg-transparent border-none shadow-none p-0 flex items-center"
+            style={{ background: 'none', border: 'none', fontWeight: 400 }}
+          >
+            Show Less
+            <svg width="12" height="12" viewBox="0 0 20 20" fill="none" className="ml-1" style={{ transform: 'rotate(180deg)' }}>
+              <title>Show fewer tags</title>
+              <path d="M6 8l4 4 4-4" stroke="currentColor" strokeWidth="2" strokeLinecap="round" strokeLinejoin="round"/>
+            </svg>
+          </button>
+        )}
       </div>
+      <div className="border-b border-gray-100 mb-8" />
+      <div className="mt-16" />
+      {/* Render agent cards: special card first, then filtered agents */}
+      {loading ? (
+        <div className="text-center text-gray-400">Loading agents...</div>
+      ) : gridAgents.length === 0 ? (
+        <div className="text-center text-gray-400">No agents found for this tag.</div>
+      ) : (
+        <div className="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-x-4 gap-y-10 px-2 md:px-10 md:max-w-[calc(100vw-200px)]">
+          {gridAgents.map((agent) => (
+            <button
+              key={agent.title}
+              type="button"
+              className="w-full text-left bg-white border border-gray-200 rounded-lg p-4 shadow-none hover:shadow-md hover:-translate-y-1 hover:bg-gray-50 transition-all"
+              onClick={() =>
+                agent.title === "Fan Segmentation Analysis"
+                  ? handleClick("wrapped")
+                  : handleAgentClick(agent)
+              }
+            >
+              <div className="text-lg mb-1 font-medium">
+                {agent.title}
+              </div>
+              <div className="text-gray-400 text-sm">{agent.description}</div>
+            </button>
+          ))}
+        </div>
+      )}
+      <div className="mb-16" />
```

Comment:
file is too long.
- required: files are `less than 100 lines of code`
- actual: file is `179 lines` (too long)

recommended: 
- new hook `useRecipes`
- new component: `<RecipeButton />`
- new component: `<RecipeTags />`
- etc